go/internal/metastore/db/dao/record_log.go

```diff
@@ -77,3 +78,40 @@ func (s *recordLogDb) PullLogs(collectionID types.UniqueID, id int64, batchSize
 		zap.Int("count", len(recordLogs)))
 	return recordLogs, nil
 }
+
+func (s *recordLogDb) GetAllCollectionsToCompact() ([]*dbmodel.RecordLog, error) {
+	log.Info("GetAllCollectionsToCompact")
+	var recordLogs []*dbmodel.RecordLog
+	var rawSql = `
+	    with summary as (
+		  select r.collection_id, r.id, r.timestamp, row_number() over(partition by r.collection_id order by r.id) as rank
+		  from record_logs r, collections c
+		  where r.collection_id = c.id
+		  and r.id>c.log_position
```

Comment:
Will compactor update the log position?

Reply:
yes go API is tracked here https://linear.app/trychroma/issue/CHR-298/segment-compaction-flush
log position needs to be updated in the same transaction that updates also registers S3 files.